ts/Extensions/Annotations/Popup.ts

```diff
@@ -932,6 +977,140 @@ H.Popup.prototype = {
+            selectBox = createElement(
+                SELECT,
+                {
+                    name: selectName,
+                    className: PREFIX + 'popup-field'
+                },
+                null as any,
+                parentDiv
+            ) as HTMLSelectElement;
+
+            selectBox.setAttribute('id', PREFIX + 'select-' + optionName);
+
+            return selectBox;
+        },
+        /**
+         * Get and add selection options.
+         *
+         * @private
+         *
+         * @param {Highcharts.AnnotationChart} chart
+         *        The chart object.
+         *
+         * @param {string} [optionName]
+         *        Name of the option into which selection is being added.
+         *
+         * @param {HTMLSelectElement} [selectBox]
+         *        HTML select box element to which the options are being added.
+         *
+         * @param {string|undefined} indicatorType
+         *        Type of the indicator i.e. sma, ema...
+         *
+         * @param {string|undefined} parameterName
+         *        Name of the parameter which should be applied.
+         *
+         * @param {string|undefined} selectedOption
+         *        Default value in dropdown.
+         *
+         * @return {void}
+         */
+        addSelectionOptions: function (
+            this: Highcharts.Popup,
+            chart: Highcharts.AnnotationChart,
+            optionName: string,
+            selectBox: HTMLSelectElement,
+            indicatorType?: string,
+            parameterName?: string,
+            selectedOption?: string
+        ): void {
+            const popup = this;
+
+            // Get and apply selection options for the possible series.
+            if (optionName === 'series') {
+                // List all series which have id - mandatory for indicator.
+                chart.series.forEach(function (serie): void {
+                    const seriesOptions = serie.options;
+
+                    if (
+                        !(seriesOptions as any).params &&
+                        seriesOptions.id &&
+                        seriesOptions.id !== PREFIX + 'navigator-series'
+                    ) {
+                        createElement(
+                            OPTION,
+                            {
+                                value: seriesOptions.id
+                            },
+                            void 0,
+                            selectBox
+                        ).appendChild(doc.createTextNode(
+                            seriesOptions.name || seriesOptions.id
+                        ));
+                    }
+                });
+            } else if (chart.options.plotOptions && indicatorType && parameterName) {
+                // Get and apply options for the possible parameters.
+                const parameterOption = pivotPointsAlgorithm;
+
+                parameterOption && parameterOption.forEach(function (element): void {
```

Comment:
Is `parameterOption &&` required here? We know that this is a nonempty array.

Even if we didn't know whether this is an array, the condition should be different. I think we can just remove `parameterOption &&`.